openedx_learning/apps/authoring/publishing/api.py

```diff
@@ -795,55 +783,54 @@ def create_next_entity_list(
     Args:
         learning_package_id: Learning package ID
         last_version: Last version of container.
-        publishable_entities_pks: The IDs of the members current members of the container.
-        entity_version_pks: The IDs of the versions to pin to, if pinning is desired.
+        entity_rows: List of ContainerEntityRows specifying the publishable entity ID and version ID (if pinned).
         entities_action: APPEND, REMOVE or REPLACE given entities from/to the container
 
     Returns:
         The newly created entity list.
     """
-    if entity_version_pks is None:
-        entity_version_pks: list[int | None] = [None] * len(publishable_entities_pks)  # type: ignore[no-redef]
     if entities_action == ChildrenEntitiesAction.APPEND:
         # get previous entity list rows
         last_entities = last_version.entity_list.entitylistrow_set.only(
             "entity_id",
             "entity_version_id"
         ).order_by("order_num")
-        # append given publishable_entities_pks and entity_version_pks
-        publishable_entities_pks = [entity.entity_id for entity in last_entities] + publishable_entities_pks
-        entity_version_pks = [  # type: ignore[operator, assignment]
-            entity.entity_version_id
+        # append given entity_rows to the existing children
+        entity_rows = [
+            ContainerEntityRow(
+                entity_pk=entity.entity_id,
+                version_pk=entity.entity_version_id,
+            )
             for entity in last_entities
-        ] + entity_version_pks
+        ] + entity_rows
     elif entities_action == ChildrenEntitiesAction.REMOVE:
         # get previous entity list rows
         last_entities = last_version.entity_list.entitylistrow_set.only(
             "entity_id",
             "entity_version_id"
         ).order_by("order_num")
-        # Remove entities that are in publishable_entities_pks
-        new_entities = [
-            entity
-            for entity in last_entities
-            if entity.entity_id not in publishable_entities_pks
+        # Remove existing children found in entity_rows
+        remove_entity_pks = [entity.entity_pk for entity in entity_rows]
+        entity_rows = [
+            ContainerEntityRow(
+                entity_pk=entity.entity_id,
+                version_pk=entity.entity_version_id,
+            )
+            for entity in last_entities.all()
+            if entity.entity_id not in remove_entity_pks
```

Comment:
Does it make sense to use `last_entities.filter(...)` here instead of iterating with a conditional?

Reply:
Yes it does! cf 84dc6f7642aa285ae378478685a22ebd2ba561b8